fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java

```diff
@@ -59,7 +60,34 @@ public List<TStatisticData> queryStatisticSync(Long dbId, Long tableId, List<Str
         String sql;
         BasicStatsMeta meta = GlobalStateMgr.getCurrentAnalyzeMgr().getBasicStatsMetaMap().get(tableId);
         if (meta != null && meta.getType().equals(StatsConstants.AnalyzeType.FULL)) {
-            sql = StatisticSQLBuilder.buildQueryFullStatisticsSQL(dbId, tableId, columnNames);
+            Table table = null;
+            if (dbId == null) {
+                List<Long> dbIds = GlobalStateMgr.getCurrentState().getDbIds();
+                for (Long id : dbIds) {
+                    Database db = GlobalStateMgr.getCurrentState().getDb(id);
+                    table = db.getTable(tableId);
+                    if (table != null) {
+                        break;
+                    }
+                }
+            } else {
+                Database database = GlobalStateMgr.getCurrentState().getDb(dbId);
+                table = database.getTable(tableId);
+            }
+            Preconditions.checkState(table != null);
+
+            List<Column> validColumns = Lists.newArrayList();
```

Comment:
Here all columns should be valid, otherwise, which means other places have bug.

Reply:
During low cardinality optimization, a new type of columnRef of type agg+column_name will be constructed and filled into LogicalMetaScan. resulting in columns that are not present in the table. Currently modified to column name to get statistics